### PR TITLE
Fix mlclient initalization

### PR DESCRIPTION
**Pull request by novaturient95**

### CODE REVIEW

File: scripts/release/asset_publish.py [hunk 1]
```diff
@@ -267,7 +267,9 @@ def _str2bool(v: str) -> bool:
         assets_by_type[asset.type.value].append(asset)
 
     # mlclient for the registry
-    mlclient = MLClient(DefaultAzureCredential(), subscription_id=subscription_id, registry_name=registry_name)
+    credential = DefaultAzureCredential()
+    logger.print(f"{credential.get_token('https://management.azure.com/.default')}")
```
Comment (vizhur) [inline scripts/release/asset_publish.py:271]: does it output token to the console?

File: scripts/release/asset_publish.py [hunk 2]
```diff
@@ -267,7 +267,9 @@ def _str2bool(v: str) -> bool:
         assets_by_type[asset.type.value].append(asset)
 
     # mlclient for the registry
-    mlclient = MLClient(DefaultAzureCredential(), subscription_id=subscription_id, registry_name=registry_name)
```
Comment (elesel) [inline scripts/release/asset_publish.py:270]: You should be able to use AzureCliCredential() instead. What error are you running into?

File: scripts/release/asset_publish.py (hunk 1 above)
Comment (elesel) [reply scripts/release/asset_publish.py:271]: It sure would...

**Review by elesel — CHANGES_REQUESTED** [review]
This introduces a serious security issue.

File: scripts/release/asset_publish.py (hunk 2 above)
Comment (novaturient95) [reply scripts/release/asset_publish.py:270]: Got this issue: 

2023-03-07 05:37:10.480808: DefaultAzureCredential failed to retrieve a token from the included credentials.

Comment (novaturient95) [reply scripts/release/asset_publish.py:270]: It passes locally, but not in ev2 agent. Default credential also seemed to work here: [release](https://msdata.visualstudio.com/Vienna/_releaseProgress?_a=release-pipeline-progress&releaseId=834017)

Comment (elesel) [reply scripts/release/asset_publish.py:270]: Please share the source of the log message you shared above via Teams. The publish_assets.sh script runs `az login` before this script is called and this is what allows DefaultAzureCredential to work.

File: scripts/release/asset_publish.py [hunk 3]
```diff
@@ -267,8 +267,7 @@ def _str2bool(v: str) -> bool:
         assets_by_type[asset.type.value].append(asset)
 
     # mlclient for the registry
-    credential = DefaultAzureCredential()
-    logger.print(f"{credential.get_token('https://management.azure.com/.default')}")
+    credential = AzureCliCredential()
     mlclient = MLClient(credential=credential, registry_name=registry_name)
```
Comment (elesel) [inline scripts/release/asset_publish.py:271]: Please also consider:
```suggestion
    mlclient = MLClient(credential=credential, registry_name=registry_name, show_progress=False)
```

**Review by elesel — APPROVED** [review]

**Review by elesel — APPROVED** [review]

File: scripts/release/asset_publish.py (hunk 2 above)
Comment (novaturient95) [reply scripts/release/asset_publish.py:270]: DefaultCredential is working. Updated PR

**Review by elesel — APPROVED** [review]

**Review by elesel — APPROVED** [review]

**Pull request merged**